### PR TITLE
DDF-1778 Remove *;resolution:=optional imports in pom files

### PR DESCRIPTION
**Pull request by jckilmer**
@coyotesqrl @roelens8 @ryeats @rzwiefel @tbatie @wbarnie 

Removed "*;resolution:=optional" imports from all pom files with the exception of ddf.catalog.transformer:catalog-responsequeuetransformer-html and ddf.test.thirdparty:rest-assured.

For the most part the required packages were explicitly imported, please comment if you think any of them should instead be embedded.

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/codice/ddf/pull/472" target="_blank">ReviewNinja

**Comment by ryeats** [comment]
:+1:

File: catalog/core/catalog-core-standardframework/pom.xml [hunk 1]
```diff
@@ -374,9 +374,17 @@
                             com.vividsolutions.jts.simplify,
                             com.vividsolutions.jts.util,
                             <!-- Remember, this is the same as saying, 2.0 or higher, not solely 2.0 -->
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-standardframework/pom.xml]: This comment should be moved back next to its referent.

**Comment by stustison** [comment]
With all of these changes, you might look into whether or not you can actually remove the Import-Package statement entirely for some of these and just let maven figure it out. That is really the ideal configuration if we can get to that point. Assuming we have the dependencies listed correctly, it should build the imports correctly as well. We need to remember that listing an import without a version next to it is basically saying that "any version will do!" and this can be dangerous in some situations (when the same package is being exported multiple times at different versions).

File: catalog/core/catalog-core-standardframework/pom.xml [hunk 2]
```diff
@@ -407,18 +418,21 @@
                             javax.xml.transform.sax,
                             javax.xml.transform.stream,
                             javax.xml.xpath,
+                            org.apache.commons.collections,
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-standardframework/pom.xml]: I don't see the `commons-collections` dependency in this jar. And, this one should be an embed once the dependency is added.

File: catalog/core/catalog-core-standardframework/pom.xml [hunk 3]
```diff
@@ -407,18 +418,21 @@
...
                             org.apache.commons.logging;version="[1.1,2)",
                             org.opengis.filter,
                             org.opengis.filter.expression,
                             org.opengis.filter.sort,
                             org.osgi.framework;version="[1.5,2)",
                             org.osgi.service.blueprint;version="[1.0.0,2.0.0)",
+                            org.osgi.service.blueprint.container,
+                            org.osgi.service.event,
                             org.slf4j;version="[1.6,2)",
+                            org.slf4j.ext,
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-standardframework/pom.xml]: The `slf4j-ext` dependency is missing from this pom.

File: platform/persistence/platform-persistence-core-impl/pom.xml [hunk 4]
```diff
@@ -186,9 +186,30 @@
...
+                            javax.naming,
+                            javax.naming.directory,
+                            javax.naming.ldap,
+                            javax.net,
+                            javax.net.ssl,
+                            javax.security.auth.x500,
+                            javax.xml.parsers,
+                            javax.xml.transform,
+                            org.apache.commons.codec.binary,
+                            org.apache.commons.lang,
```
Comment (coyotesqrl) [inline platform/persistence/platform-persistence-core-impl/pom.xml]: Probably should just embed the `commons-lang` jar.

**Comment by coyotesqrl** [comment]
Regarding @stustison's comment, please make sure that all of your dependencies are explicitly called out in each of these poms. Don't rely on transitive dependencies for artifacts that are to be embedded/packages that are to be imported into bundles.

File: catalog/core/catalog-core-standardframework/pom.xml [hunk 5]
```diff
@@ -374,9 +374,17 @@
                             com.vividsolutions.jts.simplify,
                             com.vividsolutions.jts.util,
                             <!-- Remember, this is the same as saying, 2.0 or higher, not solely 2.0 -->
+                            ddf.action,
+                            ddf.catalog,
+                            ddf.catalog.event,
+                            ddf.catalog.pubsub,
+                            ddf.catalog.transform,
                             ddf.catalog.data;version="2.0",
```
Comment (pklinef) [inline catalog/core/catalog-core-standardframework/pom.xml]: Either add `;version-"2.0",`  to the new ddf.* packages or remove it from all of the old ddf.* imports and remove the comment above.

File: catalog/ui/search-ui/search-endpoint/pom.xml [hunk 6]
```diff
@@ -388,9 +388,29 @@
                         </Private-Package>
                         <Export-Package/>
                         <Import-Package>
+                            ddf.action,
+                            ddf.catalog,
```
Comment (tbatie) [inline catalog/ui/search-ui/search-endpoint/pom.xml]: By specifying this don't you import ddf.catalog.filter, operation, source etc? Same question with ddf.security

File: platform/persistence/platform-persistence-core-impl/pom.xml [hunk 7]
```diff
@@ -179,16 +183,36 @@
...
+                            commons-lang
                         </Embed-Dependency>
                         <Embed-Transitive>true</Embed-Transitive>
                         <Private-Package>
                             org.codice.ddf.persistence.internal
                         </Private-Package>
                         <Import-Package>
+                            com.vividsolutions.jts.io,
+                            ddf.security.encryption,
+                            javax.naming,
```
Comment (tbatie) [inline platform/persistence/platform-persistence-core-impl/pom.xml]: Same question as before

File: catalog/ui/search-ui/search-endpoint/pom.xml (hunk 6 above)
Comment (jckilmer) [inline catalog/ui/search-ui/search-endpoint/pom.xml]: I think that would be done by `ddf.catalog.*`

Comment (stustison) [inline catalog/ui/search-ui/search-endpoint/pom.xml]: @jckilmer is correct. Is there a reason why we can't just do:
```
org.joda.convert;resolution:=optional,
org.eclipse.jetty.*; version="[7.6.0,9.0.0)",
*
```
Assuming the dependencies are listed correctly, this should work. I'd like to get away from listing out all of the imports manually.

Comment (jckilmer) [inline catalog/ui/search-ui/search-endpoint/pom.xml]: @stustison No reason that I know of that we can't make the changes as you suggested. When I started the ticket I was attempting to remove the `*` and `resolution:=optional` when really I could have left the `*` after making sure the dependencies were correct. I'll try that now.

**Comment by jckilmer** [comment]
@coyotesqrl You commented that dependencies should be explicitly called out, is that only for ddf dependencies? Or third party needs like osgi and apache stuff as well?

**Comment by coyotesqrl** [comment]
@jckilmer Sorry I didn't respond earlier. I was saying that if you want to embed dependencies (and, to a lesser extent import packages) in the bundle plugin that you want to make sure that those all those dependencies are explicit in the pom and not just pulled in transitively. That's true for third-party and ddf dependencies.

**Comment by jckilmer** [comment]
Thanks @coyotesqrl that was my intent on the latest set of updates.

**Comment by rzwiefel** [comment]
Full build and and started up okay! :shipit:

**Comment by roelens8** [comment]
:+1:

**Pull request merged**